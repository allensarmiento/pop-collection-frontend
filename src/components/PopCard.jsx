import Typography from '@material-ui/core/Typography';
import CardActionArea from '@material-ui/core/CardActionArea';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import { useFourThreeCardMediaStyles } from '@mui-treasury/styles/cardMedia/fourThree';
import Color from 'color';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => {
  return {
    actionArea: {
      borderRadius: 16,
      transition: '0.2s',
      '&:hover': {
        transform: 'scale(1.1)',
      },
    },
    card: ({ color }) => ({
      minWidth: 256,
      borderRadius: 16,
      boxShadow: 'none',
      '&:hover': {
        boxShadow: `0 6px 12px 0 ${Color(color)
          .rotate(-12)
          .darken(0.2)
          .fade(0.5)}`,
      },
    }),
    content: ({ color }) => {
      return {
        backgroundColor: color,
        padding: '1rem 1.5rem 1.5rem',
      };
    },
    title: {
      fontFamily: 'Keania One',
      fontSize: '2rem',
      color: '#fff',
      textTransform: 'uppercase',
    },
    subtitle: {
      fontFamily: 'Montserrat',
      color: '#fff',
      opacity: 0.87,
      marginTop: '2rem',
      fontWeight: 500,
      fontSize: 14,
    },
  };
});

const PopCard = ({ color, image, title, subtitle }) => {
  const mediaStyles = useFourThreeCardMediaStyles();
  const classes = useStyles({ color });

  return (
    <CardActionArea className={classes.actionArea}>
      <Card className={classes.card}>
        {image ? <CardMedia classes={mediaStyles} image={image} /> : (
          <CardMedia classes={mediaStyles} image={'//:0'} />
        )}
        <CardContent className={classes.content}>
          <Typography className={classes.title} variant={'h3'}>
            {title}
          </Typography>
          <Typography className={classes.subtitle}>
            {subtitle}
          </Typography>
        </CardContent>
      </Card>
    </CardActionArea>
  );
};

export default PopCard;
