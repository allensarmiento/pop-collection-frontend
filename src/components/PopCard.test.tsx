import { shallow } from 'enzyme';
import PopCard from './PopCard';

const props = {
  color: '#fff',
  image: '',
  title: '',
  subtitle: '',
};

it('renders without crashing', () => {
  expect(shallow(<PopCard {...props} />)).toMatchSnapshot();
});
