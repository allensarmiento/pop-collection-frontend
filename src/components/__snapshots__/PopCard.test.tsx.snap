// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<WithStyles(ForwardRef(CardActionArea))
  className="makeStyles-actionArea-3"
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-4 makeStyles-card-8"
  >
    <WithStyles(ForwardRef(CardMedia))
      classes={
        Object {
          "root": "makeStyles-root-1 makeStyles-root-2",
        }
      }
      image="//:0"
    />
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-content-5 makeStyles-content-9"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-6"
        variant="h3"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-subtitle-7"
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(CardActionArea))>
`;
